asyncTest
(
async
function
(
)
{
async
function
TestAsyncDisposableStackDeferOnDisposedStack
(
)
{
let
stack
=
new
AsyncDisposableStack
(
)
;
await
stack
.
disposeAsync
(
)
;
stack
.
defer
(
(
)
=
>
{
}
)
;
}
;
await
assert
.
throwsAsync
(
ReferenceError
(
)
=
>
TestAsyncDisposableStackDeferOnDisposedStack
(
)
'
Cannot
add
values
to
a
disposed
stack
!
'
)
;
}
)
;
